spec/firebase_app_distribution_action_spec.rb

```diff
@@ -1,84 +1,129 @@
-  let(:fake_upload_binary_response_valid) { double("Response", status: 202, body: { token: "projects/project_id/apps/app_id/releases/-/binaries/binary_hash" }) }
-  let(:fake_upload_status_response_success) { double("Response", status: 200, body: { status: "SUCCESS", release: { distributedAt: "1970-01-01T00:00:00Z", id: "status_id", lastActivityAt: "1970-01-01T00:00:00Z", receivedAt: "1970-01-01T00:00:00Z", displayVersion: "1.0", buildVersion: "1" } }) }
-  let(:fake_upload_status_response_invalid_app_token) { double("Response", status: 200, body: { status: "ERROR", message: "Update your Gradle plugin to >=2.0.0 or your Firebase CLI to >=8.4.1 to use the faster, more reliable upload pipeline.", release: { distributedAt: "1970-01-01T00:00:00Z", lastActivityAt: "1970-01-01T00:00:00Z", receivedAt: "1970-01-01T00:00:00Z" } }) }
+  let(:fake_connection) { double("Connection") }
   let(:fake_binary) { double("Binary") }
+  # These examples are copied directly from API responses. Maybe we should move them to another file?
+  let(:fake_get_app_response_valid) do
+    double("Response", status: 200, body: {
+      projectNumber: "project_number", appId: "app_id", platform: "android", bundleId: "bundle_id", contactEmail: "Hello@world.com"
+    })
+  end
+  let(:fake_get_app_response_no_email) do
+    double("Response", status: 200, body: {
+      projectNumber: "project_number", appId: "app_id", platform: "android", bundleId: "bundle_id", contactEmail: ""
+    })
+  end
+  let(:fake_get_app_response_invalid_app_id) do
+    double("Response", status: 400, body: {
+      error: { code: 400, message: "Request contains an invalid argument.", status: "INVALID_ARGUMENT" }
+    })
+  end
+  let(:fake_get_app_response_permission_denied) do
+    double("Response", status: 403, body: {
+      error: { code: 403, message: "The caller does not have permission", status: "PERMISSION_DENIED" }
+    })
+  end
+  let(:fake_upload_binary_response_valid) do
+    double("Response", status: 202, body: {
+      token: "projects/project_id/apps/app_id/releases/-/binaries/binary_hash"
+    })
+  end
+  let(:fake_upload_status_response_success) do
+    double("Response", status: 200, body: {
+      status: "SUCCESS", release: { distributedAt: "1970-01-01T00:00:00Z", id: "status_id", lastActivityAt: "1970-01-01T00:00:00Z", receivedAt: "1970-01-01T00:00:00Z", displayVersion: "1.0", buildVersion: "1" }
+    })
+  end
+  let(:fake_upload_status_response_invalid_app_token) do
+    double("Response", status: 200, body: {
+      status: "ERROR", message: "Update your Gradle plugin to >=2.0.0 or your Firebase CLI to >=8.4.1 to use the faster, more reliable upload pipeline.", release: { distributedAt: "1970-01-01T00:00:00Z", lastActivityAt: "1970-01-01T00:00:00Z", receivedAt: "1970-01-01T00:00:00Z" }
```

Comment:
If you keep all these fields, please format the nested objects to make it more readable:
```
..., body: {
  status: ...,
  message: ...,
  release: {
    distributedAt: ...,
    ...
  },
}
```

You could also declare the `release` in its own variable and reference it here.